tools/analyzers/ginkgolinter/ginkgo_linter.go

```diff
@@ -0,0 +1,133 @@
+package ginkgolinter
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// Analyzer is the package interface with nogo
+var Analyzer = &analysis.Analyzer{
+	Name: "ginkgolinter",
+	Doc: `enforces standards of using ginkgo and gomega
+currently, the linter searches for wrong length checks; e.g.
+	Expect(len(x).Should(Equal(1))
+This should be replaced with:
+	Expect(x).Should(HavelLen(1)`,
+	Run: run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(n ast.Node) bool {
+			// search for function calls
+			exp, ok := n.(*ast.CallExpr)
+			if ok {
```

Comment:
Maybe reverse the ok check and `continue` if `!ok` to keep the function less nested.